subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy

```diff
@@ -49,6 +49,8 @@ abstract class AbstractSmokeTest extends Specification {
 
     protected static final KotlinGradlePluginVersions KOTLIN_VERSIONS = new KotlinGradlePluginVersions()
 
+    protected static final String KGP_NO_CC_ITERATION_MATCHER = ".*(kotlin=1\\.6\\.|kotlin=1\\.7\\.).*"
+
```

Comment:
@gradle/bt-configuration-cache, this PR is currently lined up for 8.1 and I would like you to take a look.

The changes in this PR make the Kotlin Gradle Plugin < 1.8 incompatible with the configuration cache.
I marked all tests exhibiting this as such with the CC test annotations so you can find them quickly.

If we would merge this for 8.1 it would mean that Kotlin Gradle Plugin > 1.8 is required to use the configuration cache. Is this acceptable?

You might also better understand why this PR introduces this incompatibility. I have a vague idea (see the commit messages) but not sure.

Thanks

Reply:
Thanks for the heads up, @eskatos. I'll take a look to see if the impact can be reduced.